[fx] enabling typechecking of fx files

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43083 [fx] add type annotations
* **#43082 [fx] enabling typechecking of fx files**

Fixes all present errors in mypy. Does not try to add annotations everywhere.

Differential Revision: [D23145854](https://our.internmc.facebook.com/intern/diff/D23145854)

zdevito committed:
[fx] enabling typechecking of fx files

Fixes all present errors in mypy. Does not try to add annotations everywhere.

[ghstack-poisoned]

zdevito committed:
Update on "[fx] enabling typechecking of fx files"


Fixes all present errors in mypy. Does not try to add annotations everywhere.

Differential Revision: [D23145854](https://our.internmc.facebook.com/intern/diff/D23145854)

[ghstack-poisoned]

zdevito committed:
Update on "[fx] enabling typechecking of fx files"


Fixes all present errors in mypy. Does not try to add annotations everywhere.

Differential Revision: [D23145854](https://our.internmc.facebook.com/intern/diff/D23145854)

[ghstack-poisoned]

jamesr66a reviewed (approved):
LGTM, fix merge conflicts and make sure mypy runs+passes and good to go

zdevito committed:
Update on "[fx] enabling typechecking of fx files"


Fixes all present errors in mypy. Does not try to add annotations everywhere.

Differential Revision: [D23145854](https://our.internmc.facebook.com/intern/diff/D23145854)

[ghstack-poisoned]
